library/Notifications/Model/Channel.php

```diff
@@ -45,4 +47,37 @@ public function getDefaultSort()
     {
         return ['name'];
     }
+
+    public function createRelations(Relations $relations)
+    {
+        $relations->hasMany('contact', Contact::class)->setJoinType('LEFT');
+        $relations->hasMany('incident_history', IncidentHistory::class)->setJoinType('LEFT');
+        $relations->hasMany('rule_escalation_recipient', RuleEscalationRecipient::class)->setJoinType('LEFT');
+    }
+
+    /**
+     * Fetch and map all the configured channel types to a key => value array
+     *
+     * @return array<string>
+     */
+    public static function fetchChannelTypes(): array
+    {
+        $channels = [];
+        foreach (Channel::on(Database::get()) as $channel) {
```

Comment:
Please require the database connection as argument.

Reply:
I don't see the benefit of doing that.

Reply:
It's a non-testable code part less. The form already uses the singleton, so it's fine to redo it once more. But introducing the same dependency in other parts is no option.